addons/account/tests/test_chart_template.py

```diff
@@ -469,22 +470,173 @@ def get_domain(model):
+                    '__translation_module__': {
+                        'name': 'translation',
+                        'description': 'translation2',
+                    },
+                },
+            },
+            # Use 'name@' and not code translation
+            'account.tax.group': {
+                'tax_group_taxes': {
+                    'name': "Taxes",
+                    'name@fr': _translation("Taxes", "fr"),
+                    '__translation_module__': {
+                        'name': 'translation',
+                    },
+                },
+            },
+        }
+
+        def local_get_data(self, template_code):
+            data = test_get_data(self, template_code)
+            for model, record_info in translation_update_for_test_get_data.items():
+                for xmlid, data_update in record_info.items():
+                    data[model][xmlid].update(data_update)
+            return data
+
+        # Tranlations should fall back to more generic locale 'fr'
+
+        # Target lang for untranslatable fields
+        company.partner_id.lang = self.env['res.lang']._activate_lang('fr_BE').code
+
+        # Init mock translations with wrong, empty and special values
+        mock_python_translations = {
+            ('translation', 'fr'): {
+                "Taxes": "WRONG",  # Wrong translation; should not be used
+            },
+            ('no_translation', 'fr'): {},  # To test fallback to 'account'
+            ('account', 'fr_BE'): {},  # We only want to use our translations
+        }
+
+        # Add correct mock translations
+        def _add_mock_python_translation(module, lang, value):
+            mock_python_translations.setdefault((module, lang), {})[value] = _translation(value, lang)
+        _add_mock_python_translation('translation', 'fr', "Cash")
+        _add_mock_python_translation('translation', 'fr', "C")
+        _add_mock_python_translation('translation', 'fr', "Tax 1")
+        _add_mock_python_translation('translation', 'fr', "Free Account")
+        _add_mock_python_translation('translation', 'fr', "Free Tax")
+        _add_mock_python_translation('translation', 'fr', "Free Tax Description")
+        _add_mock_python_translation('translation2', 'fr', "Tax 1 Description")
+        _add_mock_python_translation('account', 'fr', "Free Account Group")
+
+        with patch.object(AccountChartTemplate, '_get_chart_template_mapping', side_effect=local_get_mapping, autospec=True):
+            with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=local_get_data, autospec=True):
+                with patch.object(AccountChartTemplate, '_post_load_data', wraps=test_post_load_data):
+                    with patch.object(code_translations, 'python_translations', mock_python_translations):
+                        self.env['account.chart.template'].try_loading('translation', company=company, install_demo=False)
+
+        # Check translations
```

Comment:
This is very complicated, which should be avoided as much as possible for tests. Business code can be complicated, but the test should be dead simple.
Can't we do something like this? (this is certainly too simple, but I'm sure you get the idea)

```python
fields_to_test = {'account.tax': ['name']}
self.assertEqual({
    f'{xmlid}.{field}@{lang}': self.env['account.chart.template'].ref(xmlid).with_context(lang=lang)[field]
    for lang in ['fr_BE', 'en_US']
    for model, data in chart_data.items():
    for xmlid in data
    for field in fields_to_test[model]
}, {
    'test_tax_1_template.name@en_US': "Tax 1",
    'test_tax_1_template.name@fr_BE': "Tax 1",
    ...
})
```